packages/lwc/index.d.ts

```diff
@@ -194,39 +194,56 @@ declare module 'lwc' {
      * @param adapter the adapter used to provision data
      * @param config configuration object for the adapter
      */
-    export function wire(
-        adapter: WireAdapterConstructor | LegacyWireAdapterConstructor,
-        config?: WireConfigValue
+    export function wire<Config extends object = any, Value = any, Context extends object = any>(
+        adapter:
+            | WireAdapterConstructor<Config, Value, Context>
+            | LegacyWireAdapterConstructor<Config, Value>,
+        config?: WireConfigValue<Config>
     ): PropertyDecorator;
 
-    type LegacyWireAdapterConstructor = (config?: any) => any;
-    type WireConfigValue = Record<string, any>;
-    type ContextValue = Record<string, any>;
+    type LegacyWireAdapterConstructor<Config = any, Value = any> = (config?: Config) => Value;
+    type WireConfigValue<Config extends object = Record<string, any>> = {
```

Comment:
This is needed for default type to maintain backward compat.

Reply:
For context: since `lwc` here is declared as ambient type in TS so everything inside here I believe it can be imported even if it is not explicitly exported so here I tried to maintain backward compat that if someone is using `WireConfigValue` somewhere it is still the same.